### PR TITLE
use `front` and `last` instead of reimplementing inline

### LINKED ISSUE #98: IterTools.pop should use Base.front instead of implementing it inline in a hackish way
This line reimplements `Base.front` in a presumably suboptimal way:

https://github.com/JuliaCollections/IterTools.jl/blob/831bbd2c584c41af961df1b9ee8954d46e366621/src/IterTools.jl#L589

I guess this would matter for the performance of bigger tuples, that is, bigger values of `k` in `subsets(collection, Val{k}())`.

The fix is simple, just use `Base.front`. However, this package seems to still support Julia 1.0, and even Julia 0.7, and `Base.front` may have been introduced after that, because I can only find it on https://docs.julialang.org starting with Julia 1.2.

### PR DESCRIPTION
**Pull request by nsajko**
Fixes #98

### CODE REVIEW

**Comment by nsajko** [comment]
bump

File: src/IterTools.jl [hunk 1]
```diff
@@ -595,7 +595,7 @@ subsets(xs::C, ::Val{K}) where {K, C} = StaticSizeBinomial{K, C}(xs)
 iterate(it::StaticSizeBinomial{0}, state=false) = state ? nothing : ((), true)
 
 # Generic case K >= 1
-pop(t::NTuple) = reverse(tail(reverse(t))), t[end]
+pop(t::NTuple) = Base.front(t), last(t)
```
Comment (oxinabox) [inline src/IterTools.jl:598]: why not
```suggestion
pop(t::NTuple) = first(t), last(t)
```
Which avoids using the internal `Base.front`  method?

Comment (nsajko) [reply src/IterTools.jl:598]: Docs link: https://docs.julialang.org/en/v1/base/collections/#Base.front

Comment (oxinabox) [reply src/IterTools.jl:598]: fair enough.
And it exists in 1.6

**Pull request merged**